### DISCUSSION TITLE
Synchronization of audio and video data

### DISCUSSION THREAD

**Discussion by PiezaKake** [bonsai-rx/bonsai#1327] [Q&A] [answered]
Hi,

I was wondering if there is the option to synchronize video and audio data online with Bonsai, without the use of a trigger

**Comment by bruno-f-cruz** [comment]
How are you acquiring the video and audio data? How precise do you need your alignment to be?

**Reply by PiezaKake** [reply]
I am using the attached file to write audio and video segments of 20s duration every time there is activity detected in the ROI. I am still playing around with the fps of the camera, but optimally would like it to be at least 250. The audio is sampled at 160 kHz.
[AudioVideoWriter.zip](https://github.com/bonsai-rx/bonsai/files/11302175/AudioVideoWriter.zip)

**Reply by bruno-f-cruz** [reply, marked as answer]
if you dont need super precise alignment you can use WithLatestFrom + ffmpeg:
https://groups.google.com/g/bonsai-users/c/gfBhLdu5tM0/m/vvk6AcVNCwAJ
